scripts/linters/custom_eslint_checks/rules/no-testonly.js

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Lint to ensure that functions with 'testOnly'
+ * in their names are only called from 'spec' files.
+ */
+
+'use strict';
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: (
+        'Functions with "testOnly" in their names should be only ' +
```

Comment:
@Hudda Should we make this a more general checker i.e, disallow-function-name-phrases, and allow .esliintrc to configure all possible names (in future we can add more such phrases? WDYT? (I'm fine with getting this PR merged and change it to a more general structure in a separate PR, let me know if needed!)

Reply:
@Hudda PTAL

Reply:
@DubeySandeep I am okay doing it in this PR if @rodemore is fine with this. And I would suggest providing 4 options:
1. to check for phrase at the starting of the function name.
2. to check for phrase at the end of the function name.
3. to check for phrase in the middle of the function name.
4. by default it will check for all positions.

wdyt?

Reply:
@Hudda 
I like the idea of having those options. I think options 1, 2 and 4 would suffice. Since it is more difficult to set the "in the middle" criteria.

Reply:
I agree with what @DubeySandeep  mentioned , it would give more flexibility to the lint and make it extensible to add more cases of terms to exclude in the name of functions. 
And I would prefer to work it in another PR to give it that focus from the beginning and make it easier for other contributors to review. But if you think it's more convenient to continue from this PR, It's fine to me.

Reply:
If you want to do this in a new PR then I would suggest that you should create an issue for this before merging this PR. Thanks!

Reply:
@Hudda  Done!  
This is the issue: #12606